src/pkg/packager/common.go

```diff
@@ -352,3 +362,117 @@ func (p *Packager) handleIfPartialPkg() error {
+
+	// Run pre-checks to make sure we have what we need to validate the checksums
+	_, err := os.Stat(p.tmp.Checksums)
+	if err != nil {
+		return fmt.Errorf("unable to validate checksums as we are unable to find checksums.txt file within the package")
+	}
+	if p.cfg.Pkg.Metadata.AggregateChecksum == "" {
+		return fmt.Errorf("unable to validate checksums because of missing metadata checksum signature")
+	}
+
+	// Get a glob of all the files within the package
+	packageGlob, err := filepath.Glob(filepath.Join(p.tmp.Base, "*"))
+	if err != nil {
+		return fmt.Errorf("unable to get a list of all the files within the package: %w", err)
+	}
+
+	// Create a map so we can track which files we have processed
+	filepathMap := make(map[string]bool)
+	for _, path := range packageGlob {
+		fileInfo, err := os.Stat(path)
+		if err != nil {
+			return fmt.Errorf("unable to get the file info for a file within the package: %s - %w", path, err)
+		}
+
+		// If the glob item is a file, add it to the map
+		if !fileInfo.IsDir() {
+			filepathMap[path] = false
+		}
+	}
+	// Verify the that checksums.txt file matches the aggregated checksum provided
+	actualChecksumSig, err := utils.GetSHA256OfFile(p.tmp.Checksums)
+	if err != nil {
+		return fmt.Errorf("unable to get the checksum of the checksums.txt file: %w", err)
+	}
+	if actualChecksumSig != p.cfg.Pkg.Metadata.AggregateChecksum {
+		return fmt.Errorf("mismatch on the checksum of the checksums.txt file, the checksums.txt file might have been tampered with")
+	}
+
+	// Check off all the files that we can trust given the checksum and signing checks
+	filepathMap[p.tmp.Checksums] = true
+	filepathMap[p.tmp.ZarfYaml] = true
+	filepathMap[p.tmp.ZarfSig] = true
+
+	// Load the contents of the checksums file
+	checksumsFile, err := os.Open(filepath.Join(p.tmp.Base, "checksums.txt"))
+	if err != nil {
+		return err
+	}
+	defer checksumsFile.Close()
+
+	/* Process every line in the checksums file */
+	scanner := bufio.NewScanner(checksumsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		// Separate the checksum from the file path
+		strs := strings.Split(scanner.Text(), " ")
+		itemPath := strs[1]
+		expectedShasum := strs[0]
+
+		actualShasum, err := utils.GetSHA256OfFile(filepath.Join(p.tmp.Base, itemPath))
+		if err != nil {
+			return err
+		}
+
+		if expectedShasum != actualShasum {
+			return fmt.Errorf("mismatch on the checksum of the %s file (expected: %s, actual: %s)", itemPath, expectedShasum, actualShasum)
+		}
+
+		filepathMap[filepath.Join(p.tmp.Base, itemPath)] = true
+	}
+
+	for path, processed := range filepathMap {
+		if !processed {
+			return fmt.Errorf("the file %s was not processed by the checksums.txt file", path)
+		}
+	}
+
+	message.Successf("All of the checksums matched!")
+	return nil
+}
+
+func (p *Packager) validatePackageSignature(publicKeyPath string) error {
+
+	// If the insecure flag we provided, ignore the signature validation
```

Comment:
```suggestion
	// If the insecure flag was provided, ignore the signature validation
```

The docs for the insecure flag should note that it skips package validation as well.